services/job_chat/tests/test_pass_fail.py

```diff
@@ -0,0 +1,449 @@
+  return { ...state, profile };
+});
+
+// Save the data
+post('https://api.example.com/save', state => ({
+  user: state.profile,
+  timestamp: new Date().toISOString()
+}));'''
+    
+    meta = {}
+    service_input = make_service_input(history=history, content=content, context=context, meta=meta, use_new_prompt=True)
+    response = call_job_chat_service(service_input)
+    print_response_details(response, "convert_get_to_post", content=content)
+    
+    assert response is not None
+    assert "response" in response
+    assert "suggested_code" in response
+    
+    assert response["suggested_code"] == expected_code or response["suggested_code"] == expected_code_alternative, f"GET to POST conversion did not produce expected result.\nExpected:\n{expected_code}\n\nActual:\n{response['suggested_code']}"
+
+def test_rename_function():
+    print("==================TEST==================")
+    print("Description: Testing function name change in longer OpenFn workflow")
+    
+    history = []
+    content = "Rename the 'getPatientData' function to 'fetchPatientRecords'"
+    
+    context = {
+        "expression": '''// Custom helper functions
+function getPatientData(patientId) {
+  return `https://hospital.org/api/patients/${patientId}`;
+}
+
+function formatDate(date) {
+  return new Date(date).toISOString().split('T')[0];
+}
+
+// Initial configuration
+fn(state => {
+  const config = {
+    apiVersion: '2.1',
+    requestTimeout: 30000,
+    maxRetries: 3
+  };
+  return { ...state, config };
+});
+
+// Fetch data for each patient
+each(
+  '$.patients[*]',
+  fn(state => {
+    console.log(`Processing patient ID: ${state.data.id}`);
+    return state;
+  }),
+  get(state => getPatientData(state.data.id)),
+  fn(state => {
+    const patient = state.data;
+    // Enrich patient data
+    patient.lastUpdated = formatDate(new Date());
+    patient.dataSource = 'API';
+    patient.endpoint = getPatientData(patient.id);
+    
+    return { ...state, patient };
+  })
+);
+
+// Process the results
+fn(state => {
+  const patients = state.references.map(ref => ref.patient);
+  console.log(`Retrieved data for ${patients.length} patients`);
+  
+  const summary = {
+    count: patients.length,
+    source: getPatientData('summary'),
+    retrievalDate: formatDate(new Date())
+  };
+  
+  return { ...state, patients, summary };
+});
+
+// Save individual patient records
+each(
+  '$.patients[*]',
+  post(
+    'https://records.hospital.org/store',
+    state => ({
+      patient: state.data,
+      metadata: {
+        source: getPatientData(state.data.id),
+        processedAt: new Date().toISOString()
+      }
+    })
+  )
+);
+
+// Submit the final report
+post(
+  'https://reports.org/submit', 
+  state => ({
+    source: 'Patient API',
+    results: state.patients,
+    summary: state.summary,
+    endpoint: getPatientData('summary')
+  })
+);'''
+    }
+    
+    expected_code = '''// Custom helper functions
```

Comment:
Presumably, rather than duplicating all this code, you could just do a simple string replace on the function name? That also has the benefit of being easier to read for humans. Because I, as a test reviewer, can't easily tell you the difference between the original and expected code snippets